File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_unroll_runme.go
package main

import "./director_unroll"

type MyFoo struct{} // From director_unroll.Foo
func (p *MyFoo) Ping() string {
	return "MyFoo::ping()"
}

func main() {
	a := director_unroll.NewDirectorFoo(&MyFoo{})

	b := director_unroll.NewBar()

	b.Set(a)
	c := b.Get()

	if c.Ping() != "MyFoo::ping()" {
		panic(c.Ping())
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/director_unroll_runme.pl
use strict;
use warnings;
use Test::More tests => 3;
BEGIN { use_ok 'director_unroll' };
require_ok 'director_unroll';

{
  package MyFoo;
  use base 'director_unroll::Foo';
  sub ping { "MyFoo::ping()" }
}

$a = MyFoo->new();
$b = director_unroll::Bar->new();
$b->set($a);
my $c = $b->get();
is(${$a->this}, ${$c->this}, "unrolling");
